searchcore/src/vespa/searchcore/proton/matching/attribute_limiter.cpp

```diff
@@ -76,19 +76,19 @@ toString(AttributeLimiter::DiversityCutoffStrategy strategy)
     return (strategy == AttributeLimiter::DiversityCutoffStrategy::STRICT) ? STRICT_STR : LOOSE_STR;
 }
 
-std::unique_ptr<SearchIterator>
-AttributeLimiter::create_search(size_t want_hits, size_t max_group_size, double hit_rate, bool strictSearch)
-{
+search::fef::MatchData &
```

Comment:
consider returning a pair of blueprint ref and match data ref to not have to touch the internal blueprint without the lock afterwards.

Reply:
Fixed